solr/modules/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -241,10 +242,31 @@ public void setContext(ResultContext context) {
      *
      * @param transformerFeatureStore the explicit transformer feature store
      */
-    private LoggingModel createLoggingModel(String transformerFeatureStore) {
-      final ManagedFeatureStore fr = ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+    private LoggingModel createLoggingModel(
+        String transformerFeatureStore, Boolean docsWereNotReranked) {
+      final ManagedFeatureStore featureStore =
+          ManagedFeatureStore.getManagedFeatureStore(req.getCore());
+
+      // check for non-existent feature store name
+      if (transformerFeatureStore != null) {
+        if (!featureStore.getStores().containsKey(transformerFeatureStore)) {
```

Comment:
do we need this? isn't this already in the  'final FeatureStore store = featureStore.getFeatureStore(transformerFeatureStore);' ?

Do we even need 'featureStore.getStores()' ?

Reply:
When a nonexistent feature store is passed with _"final FeatureStore store = featureStore.getFeatureStore(transformerFeatureStore);"_ no error is received and the store is improperly added (empty) to the list of feature stores. In fact, the getFeatureStore() method is also used for the 'put' phase.

So a check was added to get the error and avoid creating it.

Reply:
mmmm what is the difference with:
`public synchronized FeatureStore getFeatureStore(String name) {
     if (name == null) {
       name = FeatureStore.DEFAULT_FEATURE_STORE_NAME;
 @@ -160,11 +164,12 @@ public void doGet(BaseSolrResource endpoint, String childId) {
     if (childId == null) {
       response.add(FEATURE_STORE_JSON_FIELD, stores.keySet());
     } else {
       // check for non-existent feature store name
       if (!stores.containsKey(childId)) {
         throw new SolrException(
             SolrException.ErrorCode.BAD_REQUEST, "Missing feature store: " + childId);
       }` ?

Reply:
the formatting is terrible but you should get the idea

Reply:
If I have understood correctly your doubt... 
I added two checks because _doGet_ method is used when you want to get a feature store, for example:
`GET http://localhost:8983/solr/books/schema/feature-store/nonExistentFeatureStore`

While _createLoggingModel()_ method is used when you want to extract features, for example:
`GET http://localhost:8983/solr/books/query?q=test&fl=id,[features store=nonExistentFeatureStore]
`

Here I also handled the cases when the store name is not passed (i.e., is null [features]) and when the rerank query is present.